Meeting 06.md (#344)
ethereum/pm: Merge/Meeting 06.md

# Merge Implementers' Call #6 Notes

### Meeting Date/Time: Thursday 2021/6/17 at 13:00 UTC
### Meeting Duration: 35 minutes
### [GitHub Agenda](https://github.com/ethereum/pm/issues/340)
### [Audio/Video of the meeting](https://youtu.be/b5gh0Mw2oPU)
### Moderator: Mikhail Kalinin
### Notes: Santhosh(Alen)

# Agenda
- Implementation updates
- Research updates
  - Polish merge/beacon-chain.md, ethereum/eth2.0-specs#2472
  - Add randao to execution payload, ethereum/eth2.0-specs#2479
- Spec discussion
  - Location of execution-layer specs
- Open discussions

# Intro
**Mikhail Kalinin**
* Let's just get started yeah welcome to the Merge Implementers' Call #6 today is going to be pretty straightforward I guess so we'll start from implementation updates as usual there is yeah there is update from my side i'm working on the transition process implementation attacker because the pr has been merged into the into this big repo so I feel like i'll finish this implementation like t the end of this week or early next week and we'll try to.

> * Yeah, the next challenge will be to set up the proofwork chain locally around the beacon chain, do all the steps that we did on the mainnet already, and finish it with the merge transition process, so probably I'll just you yeah yeah likely reuse the scripts that we have from after rayonism, which is great, and yeah we'll use the Gary PR, which already contains everything.

Reviewer: ```suggestion
* Yeah, the next challenge will be to set up the proof of work chain locally around the beacon chain, do all the steps that we did on the mainnet already, and finish it with the merge transition process, so probably I'll just you yeah yeah likely reuse the scripts that we have from after rayonism, which is great, and yeah we'll use the Gary PR, which already contains everything.
```

* The transition logic and in gaf so we'll see this is my update does anyone else have an implementation update yep I believe it makes sense everyone is working on the corresponding hard forks so okay cool any questions here.
* Okay, let's go on to the research updates. We have a couple of things to highlight and possibly discuss, and we have Justin on the line I guess 

# [Research Discussion](https://youtu.be/b5gh0Mw2oPU&t=70s)

**Danny**

> * I can provide some context. I'm talking about 24/72. Justin is known for going through polishing and merging making sure that the specs conform in terms of naming conventions and structure and all that kind of stuff, so he's done that on the recent merge specs has been about a bunch of review I just did it's now passing ci and and has incorporated the feedback so I gave it a plus one this morning if anyone wants to take a look at it.

Reviewer: ```suggestion
* I can provide some context. I'm talking about PR [2472](ethereum/eth2.0-specs#2472). Justin is known for going through polishing and merging making sure that the specs conform in terms of naming conventions and structure and all that kind of stuff, so he's done that on the recent merge specs has been about a bunch of reviews I just did it's now passing ci and and has incorporated the feedback so I gave it a plus one this morning if anyone wants to take a look at it.
```

**Danny**
* I mean, my point is that we have to modify difficulty since it requires the new context, and so defining it as a constant or taking it as the value off of that rpc uh has probably small complexity difference, so we might as well do it at all.

**Mikhail Kalinin**
* At this point, the question Yes, that is the question, so is it a huge concern that you have to transmit it directly to the evm without any intermediate steps?

**Danny**
* Yeah, the execution layer gets context and directives, so I don't think it's a major concern, but I'm not in charge of the program on that side.

**Mikhail Kalinin**
* yeah, this is the question, particularly for net client monitors, how tough is it to adjust the.

**Danny**
* Well, that will have to be adjusted again. It could be something like if post-merge difficulty equals one um or if post-merge difficulty equals this value that has been passed on.

**Micah Zoltu**

> * Would appreciate hearing from Tommaso or Ryan.

Reviewer: ```suggestion
* Would appreciate hearing from Tomasz or Ryan.
```

**Tim Beiko**
* Okay, that sounds good; I'll follow up with you and nikki allen; others are attempting to acquire a first draft of it.

**Danny**
* Great yeah.

**Mikhail Kalinin**
* I believe we require a timeline with checkboxes.

**Tim Beiko**
* Yeah, I'm arguing for checkboxes to avoid the timeline, and I believe we did this fairly well with 1559 where we have this checklist because there will be increasing pressure at the worst time when stuff is like 50 ready and people will start asking you know when merge and being able to say well look you know here are like 10 things we still need to figure out.
* You know, both for us, I think first and foremost, but also for the community, there's value in seeing like, oh, the consensus changes are done but sync is broken or json rpc is broken or whatever right yeah, I definitely would not put dates on that docent and kind of use it as a shield against having to provide dates.

**Mikhail Kalinin**

> * okay see yeah okay as for the eiep process, I don't think it makes much sense to put every different part of the execution client change into a separate dip probably we can use the approach that was taken in the click eip which just describes everything in one docent what do you think I mean once we figure out the synth process once we figure out yeah seeing process definitely will be.

Reviewer: ```suggestion
* okay see yeah okay as for the eip process, I don't think it makes much sense to put every different part of the execution client change into a separate dip probably we can use the approach that was taken in the click eip which just describes everything in one docent what do you think I mean once we figure out the synth process once we figure out yeah seeing process definitely will be.
```

**Danny**
* Yep, we have some fork integration tests for altair from phaser altair, so we can at least use some of that as a basis, but we'll have to work out how we exactly integrate and or stub the execution side in those tests.

**Mikhail Kalinin**
* Oh okay thanks everyone i'll see you in 25 minutes thanks everyone thank you what's this the call that we were talking over um api last week two weeks ago or is that e2 call if you call good.

-------------------------------------------
## Speaking Attendees

**Lukasz Rozmej**  
**Pooja | ECH**  
**Gary Schulte**  
**Mikhail Kalinin**  
**Micah Zoltu**  
**Dustin Brody**  
**Terence**  
**Danny**  
**Jacek Sieka**  
**Paul Hauner**  
**Alex stokes**  
**Vitalik**  
**Dankrad Feist Kevaundray Wedderburn**  
**DCinvestor**  
**Ben Edgington**  
**Protolambda**  
**Sajida Zouarhi**  
**Hsiao-Wei Wang**  
**Karim T**  
**Kristof Gazso**  
**Adrian Sutton**  
**Bhargavasomu**  
**Tim Beiko**  
**Trenton Van Epps**  
**Mamy**  
**Tomasz Stanczak**  
**Ansgar Dietrichs**  

> 

Reviewer: ```suggestion
## Next Call - July 01, 2021 at 1300 UTC
```
